opencog/atoms/distvalue/DistributionalValue.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * opencog/truthvalue/DistributionalValue.h
+ *
+ * Copyright (C) 2018 SingularityNet
+ * All Rights Reserved
+ *
+ * Written by Roman Treutlein
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _OPENCOG_DISTRIBUTIONAL_VALUE_H
+#define _OPENCOG_DISTRIBUTIONAL_VALUE_H
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <limits>
+
+#include <opencog/util/Counter.h>
+#include <opencog/atoms/distvalue/ConditionalDV.h>
+
+/** \addtogroup grp_atomspace
+ *	@{
+ */
+
+namespace opencog
+{
+
+typedef double strength_t;
+typedef double confidence_t;
+typedef double count_t;
+
+class DistributionalValue;
+
+typedef std::vector<double> DVec;
+
+typedef std::vector<double> Interval;
+
+typedef std::vector<Interval> DVKey;
+
+typedef std::vector<DVKey> DVKeySeq;
+
+typedef Counter<DVKey, double> DVCounter;
+
+typedef std::shared_ptr<const DistributionalValue> DistributionalValuePtr;
+
+class AtomSpace;
+
+class DistributionalValue
+	: public Value
+{
+
+	friend class ConditionalDV;
+
+	DVCounter _value;
+
+	// Disallow assignment -- truth values are immutable!
+	DistributionalValue& operator=(const DistributionalValue& rhs) {
+		throw RuntimeException(TRACE_INFO, "Cannot modify truth values!");
+	}
+
+
+public:
+	static count_t DEFAULT_K;
+
+	DistributionalValue();
+	DistributionalValue(const DVCounter&);
```

Comment:
What do the values of DVCounter represent? According to get_mean and get_mode they represent parameters of a Dirichlet distribution. But in the absence of comments the user might think that they represent evidence counts.

@rTreutlein, you need to better comment your code. My personal policy regarding comments is as follows
1. Put the "what" in header file. Everything the user needs to know to use the code. What each function does, using math formulas if you need to, pointing to wikipedia pages, articles, github issues if you need to, adding examples if you need to. The user should be able to read your comments and know what it is doing without having to read the code.
2. Put the "how" in the source file. Everything the user may not care but the developer/maintainer may need.

See my last piece of code, the pattern miner, as examples of comments https://github.com/opencog/opencog/blob/master/opencog/learning/miner/Miner.h https://github.com/opencog/opencog/blob/master/opencog/learning/miner/MinerUtils.h (it's not a coincidence that my example comments are in my last code, that's because I've come to learn over the years how important good comments are).

You should also add a README.md, see for instance https://github.com/opencog/opencog/blob/master/opencog/learning/miner/README.md

Writing good comments and good documentation takes time, but it's just as important as good code.